### DISCUSSION TITLE
How to use `useCookie` composable right way in server api?

### DISCUSSION THREAD

**Discussion by akagire** [nuxt/nuxt#17192] [Questions] [answered]
Hello. I'm trying to use `useCookie` with Nuxt server in `defineEventHandler` .

I read [unijs/h3's JSDoc](https://www.jsdocs.io/package/h3#useCookie) and I wrote following code.

```ts
export default defineEventHandler(event => {
  const test = useCookie(event, 'test')
  return {
    test
  }
})
```

This code, I got ts error in `event`.

<img width="915" alt="スクリーンショット 2022-07-27 18 32 13" src="https://user-images.githubusercontent.com/9305065/181214111-6aa03230-070f-4f93-b5aa-c7866f3df2f0.png">

But this code is working. When I remove `event` , ts error is gone, but not working with following error message.

```
[nuxt] [request error] Cannot read properties of undefined (reading 'headers')
  at useCookies (./node_modules/h3/dist/index.mjs:155:26)  
  at useCookie (./node_modules/h3/dist/index.mjs:158:10)  
  at ./.nuxt/dev/index.mjs:414:16  
  at ./node_modules/h3/dist/index.mjs:311:47  
  at processTicksAndRejections (node:internal/process/task_queues:96:5)  
  at async ./node_modules/h3/dist/index.mjs:420:19  
  at async nodeHandler (./node_modules/h3/dist/index.mjs:370:7)  
  at async ufetch (./node_modules/unenv/runtime/fetch/index.mjs:9:17)  
  at async $fetchRaw2 (./node_modules/ohmyfetch/dist/chunks/fetch.mjs:131:20)  
  at async setup (./.nuxt/dist/server/server.mjs:2897:101)
```

Currently, I use `useCookie` with `@ts-ignore` comment.

Please tell how to use `useCookie` right way in server side for current version if you know.

Or, you know this composables type definition is maybe expected use in Vue SFC file, so I'm wondering we need to prepare for server side `useCookie` type definition.

Thanks for your help.

**Comment by Diizzayy** [comment, marked as answer]
> This code, I got ts error in event.

@Akagire It appears that your typescript intellisense is resolving to the Nuxt 3 counterpart of `useCookie`, simply adding the import from h3 should remove the error 

`import { useCookie } from 'h3'`

**Comment by akagire** [comment]
@Diizzayy Thanks for your help and I resolved!
